### PR TITLE
#55 Call timeline filter after simulation ended

### LINKED ISSUE #55: Call timeline filter after the simulation
Automatically filter the potential timelines generated during the simulations after the process finished

### PR DESCRIPTION
**Pull request by rosiereflo**
closes #55

### CODE REVIEW

File: sources/execDynawoAlgorithms.sh [hunk 1]
```diff
@@ -77,12 +77,43 @@ setEnv() {
   export LD_LIBRARY_PATH=$DYNAWO_ALGORITHMS_INSTALL_DIR/lib:$LD_LIBRARY_PATH
 }
 
+find_and_call_timeline() {
+  if [ ! -d "$1" ]; then
+    return 1
+  fi
+  find $1 -name $2 | while read line; do
```
Comment (gautierbureau) [inline sources/execDynawoAlgorithms.sh:84]: line variable is a file ?

Comment (rosiereflo) [reply sources/execDynawoAlgorithms.sh:84]: not sure I get the question, this line looks like this for example: find /home/jobsfolder/ -name timeline.log

Comment (rosiereflo) [reply sources/execDynawoAlgorithms.sh:84]: oh ok I got it, yes line is a file do you want me to update the name?

Comment (valleefred) [reply sources/execDynawoAlgorithms.sh:84]: line is a line in the result lines of command find, so yes, line is a filename !

File: sources/execDynawoAlgorithms.sh [hunk 2]
```diff
@@ -77,12 +77,43 @@ setEnv() {
...
+    echo "Processing file '$line'"
+    python $DYNAWO_ALGORITHMS_INSTALL_DIR/bin/timelineFilter.py --timelineFile $line
```
Comment (valleefred) [inline sources/execDynawoAlgorithms.sh:86]: Maybe you should use $DYNAWO_ALGORITHMS_PYTHON_COMMAND

File: util/envDynawoAlgorithms.sh [hunk 3]
```diff
@@ -599,6 +599,7 @@ create_distrib() {
   cp ../bin/* bin/
   cp ../lib/* lib/.
   cp -r ../share/* share/
+  cp $DYNAWO_HOME/sbin/timeline_filter/timelineFilter.py bin/.
```
Comment (valleefred) [inline util/envDynawoAlgorithms.sh:602]: Why timeline filter is not called after simulation ended in development environment as in execution environment ?

**Review by valleefred — CHANGES_REQUESTED** [review]

File: sources/execDynawoAlgorithms.sh (hunk 1 above)
Comment (gautierbureau) [reply sources/execDynawoAlgorithms.sh:84]: Yes my comment was not clear but if the variable line is indeed a file better name filename I think

Comment (rosiereflo) [reply sources/execDynawoAlgorithms.sh]: fixed

File: sources/execDynawoAlgorithms.sh (hunk 2 above)
Comment (rosiereflo) [reply sources/execDynawoAlgorithms.sh]: It is not defined in the execution environment

File: util/envDynawoAlgorithms.sh (hunk 3 above)
Comment (rosiereflo) [reply util/envDynawoAlgorithms.sh:602]: It has to be called in the platform to ease readability of file, as the development environment is for developers they can call directly the script if needed. If we see it is required we can add it later

**Review by valleefred — APPROVED** [review]

**Review by gautierbureau — APPROVED** [review]

**Pull request merged**